storage/uri.go

```diff
@@ -59,53 +61,131 @@ func (u *uri) String() string {
 	return u.raw
 }
 
+// parentGeneric is a generic function that returns the last element of a
+// path after splitting it on "/". It should be suitable for most URIs.
+func parentGeneric(location string) (string, error) {
+
+	// trim leading forward slashes
+	trimmed := 0
+	for location[0] == '/' {
+		location = location[1:]
+		trimmed++
+
+		// if all we have left is an empty string, than this URI
+		// pointed to a UNIX-style root
+		if len(location) == 0 {
+			return "", URIRootError
+		}
+	}
+
+	components := strings.Split(location, "/")
+
+	if len(components) == 1 {
+		// trimmed <= 2 makes sure we handle UNIX-style paths on
+		// Windows correctly
+		return "", URIRootError
+	}
+
+	parent := ""
+	if trimmed > 2 && len(components) > 1 {
+		// Because we trimmed all the leading '/' characters, for UNIX
+		// style paths we want to insert one back in. Presumably we
+		// trimmed two instances of / for the scheme.
+		parent = parent + "/"
+	}
+	parent = parent + strings.Join(components[0:len(components)-1], "/") + "/"
+
+	return parent, nil
+}
+
 // Parent gets the parent of a URI by splitting it along '/' separators and
 // removing the last item.
 func Parent(u fyne.URI) (fyne.URI, error) {
 	s := u.String()
 
 	// trim trailing slash
+	if runtime.GOOS == "windows" {
+		if s[len(s)-1] == '\\' {
+			s = s[0 : len(s)-1]
+		}
+	}
```

Comment:
Should this not be in an else? Would a windows path with / at the end  expect to have it truncated?

Reply:
I removed the check for windows since all URIs now have ```/```

Reply:
Yes, I think a trailing `\` is no longer relevant if we are normalizing everything to `/`.